src/App.tsx

```diff
@@ -43,33 +54,88 @@ type State = {
   sortType: SortType,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  startGoods = () => {
+    this.setState({ isStarted: true });
+  };
+
+  sortByType = (type: SortType) => {
+    this.setState({ sortType: type });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => ({ isReversed: !state.isReversed }));
+  };
+
+  resetGoodsOrder = () => {
+    this.setState({ isReversed: false, sortType: 0 });
```

Comment:
Don't use indeces for `enum` values, you have more clear, word definition
```suggestion
    this.setState({ isReversed: false, sortType: SortType.NONE });
```